pyanaconda/modules/storage/devicetree/model.py

```diff
@@ -266,15 +273,17 @@ def reset(self, cleanup_only=False):
 
         # Protect devices from teardown.
         self._mark_protected_devices()
-        self.devicetree.teardown_all()
+        if deep:
+            self.devicetree.teardown_all()
```

Comment:
Btw. we also call `teardown_all` during the installation, after we unlock a device and after we discover new devices. If I understand it correctly, Anaconda can never tear down any device if Cockpit Storage was used for the partitioning, right?

Reply:
I think that `teardown_all` should be disabled globally via a flag or something if it is requested. It will be safer, more predictable and cleaner. You can start with using `devicetree.teardown_all = lambda: log.debug("Tear down is disabled.")` as a temporary workaround for testing.